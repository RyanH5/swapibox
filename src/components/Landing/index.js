import React, { Component } from 'react';
import { getData } from '../../helpers/selectCategory/index';
import './styles.css';

class Landing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      year: '',
      crawl: ''
    };
  }

  async componentDidMount() {
    const randomNum = Math.floor(Math.random() * 7) + 1;
    const filmData = await getData(`https://swapi.co/api/films/${randomNum}`);
    const {title, year, crawl} = filmData;
    this.setState({crawl, title, year});
  }

  render() {
    const { title, year, crawl } = this.state;
    return (
      <div className={'landing'}>
        <div className={"fade"}></div>
        <main>
          <div className={'crawl'}>
            <h2>{title}</h2>
            <p>{crawl}</p>
            <h4>{year}</h4>
          </div>
        </main>
      </div>
    );
  }
}

export default Landing;