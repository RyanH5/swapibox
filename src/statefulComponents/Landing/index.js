import React, { Component } from 'react';
import { fetchFilmCrawl } from '../../helpers/fetch/index';

class Landing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      year: '',
      crawl: ''
    };
  }

  async componentDidMount() {
    const filmData = await fetchFilmCrawl();
    const {title, year, crawl} = filmData;
    this.setState({crawl, title, year});
  }

  render() {
    const { title, year, crawl } = this.state;
    return (
      <div>
        <h2>{title}</h2>
        <p>{crawl}</p>
        <h4>{year}</h4>
      </div>
    );
  }
}

export default Landing;