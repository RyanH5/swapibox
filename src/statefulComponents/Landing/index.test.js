import React from 'react';
import Landing from './index';
import { shallow } from 'enzyme';

describe('Landing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Landing />, {disableLifecycleMethods: true});
  });

  it('should display Landing with correct data', async () => {
    expect(wrapper).toMatchSnapshot();
  });

});