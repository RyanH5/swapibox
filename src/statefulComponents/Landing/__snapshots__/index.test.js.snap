// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should display Landing with correct data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <p />,
        <h4 />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <p />,
          <h4 />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
